### PR TITLE
update the algorithm for triggering pods status for deployments

### PR DESCRIPTION
**Pull request by mohamedveron**

### CODE REVIEW

File: executor/status_handler.go [hunk 1]
```diff
@@ -107,19 +108,34 @@ func (executor *Executor) podsStatusHandler(entityName string, namespace string,
 func (executor *Executor) deploymentsHandler(entityName string, namespace string) (deploymentName string, targetPods int32, err error) {
 	replicasets, err := executor.kube.GetNamespaceReplicaSets(namespace)
 
-	if err != nil {
```
Comment (alichaddad) [inline executor/status_handler.go:110]: Why was this removed

**Review by iahvector — CHANGES_REQUESTED** [review]

File: executor/status_handler.go (hunk 1 above)
Comment (mohamedveron) [reply executor/status_handler.go:110]: no need to double check for the replicaset name, we fetch the right one before the short pooling loop, so every time we will get the same replicaset so no need for this additional request

**Review by iahvector — APPROVED** [review]

**Review by alichaddad — APPROVED** [review]
